tests/antool/test-fi.sh

```diff
@@ -0,0 +1,290 @@
+	;;
+12)	# corrupt the signature of vltrace log
+	truncate -s 10 $BIN_LOG
+	# append 2 bytes
+	echo -n "__" >> $BIN_LOG
+	;;
+13)	# corrupt the version of vltrace log (set it to 0.0.0)
+	truncate -s 12 $BIN_LOG
+	# append 12 null bytes
+	echo -n -e '\x0\x0\x0\x0\x0\x0\x0\x0\x0\x0\x0\x0' >> $BIN_LOG
+	;;
+14)	# corrupt the architecture of vltrace log (set it to 0)
+	truncate -s 24 $BIN_LOG
+	# append 4 null bytes
+	echo -n -e '\x0\x0\x0\x0' >> $BIN_LOG
+	;;
+15)	# corrupt the architecture of vltrace log (set it to very large number)
+	truncate -s 24 $BIN_LOG
+	# append 4 bytes
+	echo -n "abcd" >> $BIN_LOG
+	;;
+16)	# corrupt the size of the syscalls table entry
+	truncate -s 28 $BIN_LOG
+	# append 4 bytes
+	echo -n "abcd" >> $BIN_LOG
+	;;
+#
+# cut out or move packets
+#
+17)	# move 2 packets to the end of log
+	move_part_to_end $BIN_LOG 80755 80883
+	ANTOOL_OPTS="-d"
+	;;
+18)	# move 2 packets to the end of log
+	move_part_to_end $BIN_LOG 80755 80883
+	ANTOOL_OPTS="-d -f"
+	;;
+19)	# cut out 2 packets from the log
+	cut_out_from_file $BIN_LOG 80755 80883
+	ANTOOL_OPTS="-d"
+	;;
+20)	# cut out 2 packets from the log
+	cut_out_from_file $BIN_LOG 80755 80883
+	ANTOOL_OPTS="-d -f"
+	;;
+21)	# move first packet of 'open' to the end of log
+	move_part_to_end $BIN_LOG 88863 89331
+	ANTOOL_OPTS="-d"
+	;;
+22)	# cut out some packets from the log in order to get
+	# the 'list_no_exit' and 'list_others' lists not empty
+	cut_out_from_file $BIN_LOG 80755 80883
+	cut_out_from_file $BIN_LOG 149515 150919
+	cut_out_from_file $BIN_LOG 181951 181995
+	cut_out_from_file $BIN_LOG 216599 216643
+	cut_out_from_file $BIN_LOG 220771 220815
+	ANTOOL_OPTS="-d -f -l"
+	;;
+23)	# zero pid_tid of packet in STATE_IN_ENTRY state of SyS_open
+	replace_n_bytes $BIN_LOG 89340 8 '\x0\x0\x0\x0\x0\x0\x0\x0'
+	ANTOOL_OPTS="-s -d"
+	;;
+24)	# inject BPF read error in the 1st packet of open (offset 81567)
+	replace_n_bytes $BIN_LOG 81573 1 '\x4' # offset +6
+	replace_n_bytes $BIN_LOG 81648 1 '\x0' # offset +81
+	ANTOOL_OPTS="-m 12 -s -d"
+	;;
+25)	# inject BPF read error in the 1st packet of open (offset 81567)
+	replace_n_bytes $BIN_LOG 81573 1 '\x4' # offset +6
+	replace_n_bytes $BIN_LOG 81648 1 '\x0' # offset +81
+	ANTOOL_OPTS="-m 12 -s -f -l"
+	;;
+26)	# change size of SyS_open packet (offset 81567)
+	replace_n_bytes $BIN_LOG 81568 2 '\x4B\x00'
+	ANTOOL_OPTS="-m 9 -s -d"
+	;;
+27)	# change size of SyS_fork packet (offset 87711)
+	replace_n_bytes $BIN_LOG 87712 2 '\x4B\x00'
+	ANTOOL_OPTS="-m 87 -s"
+	;;
+esac
+
+ANTOOL_OPTS="-b $BIN_LOG $ANTOOL_OPTS"
+
+set +e
+echo "$ $ANTOOL $ANTOOL_OPTS > $OUT 2> $ERR"
+$ANTOOL $ANTOOL_OPTS > $OUT 2> $ERR
+RV=$?
+set -e
+
+if [ $RV -ne $EXPECTED_RV ]; then
+	echo "Error: return value ($RV) different than expected ($EXPECTED_RV)"
+	exit 1
+fi
+
+check
+
+# test succeeded
+
+# copy coverage report
+[ "$COVERAGE" == "1" -a -f $COVERAGE_REPORT ] && cp -f $COVERAGE_REPORT ..
+
+# remove the temporary test directory
+cd ..
+rm -rf $DIR_NAME
```

Comment:
Please remove files without using "-rf" flag. It's very dangerous. Also please validate that all variables used in rm are not empty.

Reply:
Done